runtime/dart_isolate.cc

```diff
@@ -803,10 +799,22 @@ DartIsolate::CreateDartVMAndEmbedderObjectPair(
 // |Dart_IsolateShutdownCallback|
 void DartIsolate::DartIsolateShutdownCallback(
     std::shared_ptr<DartIsolate>* embedder_isolate) {
+  // Terminate child Isolates
+  if ((*embedder_isolate)->terminate_child_isolates_) {
+    for (std::weak_ptr<DartIsolate> child_isolate :
+         (*embedder_isolate)->child_isolates_) {
+      if (!child_isolate.expired() && child_isolate.lock() != nullptr) {
+        bool shutdown_success = child_isolate.lock()->Shutdown();
+        if (!shutdown_success) {
+          FML_DLOG(ERROR) << "Child Isolate failed to shutdown properly.";
+        }
+      }
+    }
+  }
```

Comment:
Please be wary of cyclomatic complexity. We prefer short methods with early returns. So something like `TerminateChildIsolatesIfNecessary` mentioned above with early returns in case the child isolates need not be terminated.